api/status_info.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+)
+
+// StatusInfo contains information about a user's status on a particular language track.
+type StatusInfo struct {
+	TrackID         string `json:"track_id"`
+	Recent          *Recent
+	FetchedProblems []string `json:"fetched"`
+	SkippedProblems []string `json:"skipped"`
+}
+
+// Recent contains information about the user's most recently submitted exercise on a particular language track.
+type Recent struct {
+	Problem     string `json:"problem"`
+	SubmittedAt string `json:"submitted_at"`
+}
+
+func (r *Recent) String() string {
+	submittedAt, err := time.Parse(time.RFC3339Nano, r.SubmittedAt)
+	if err != nil {
+		log.Fatal(err)
+	}
+	year, month, day := submittedAt.Date()
+
+	return fmt.Sprintf(" - %v (submitted on %v %v, %v)", r.Problem, month, day, year)
+}
+
+func (s *StatusInfo) String() string {
+	if len(s.FetchedProblems) == 0 && s.Recent.Problem == "" {
+		return fmt.Sprintf("\nYou have yet to begin the %v track!\n", s.TrackID)
+	}
+
+	msg := `
+Your status on the %v track:
+
+Most recently submitted exercise:
+%v
+
+Exercises fetched but not submitted:
+%v
+
+Exercises skipped:
+%v
+`
+
+	return fmt.Sprintf(msg, s.TrackID, s.Recent, formatProblems(s.FetchedProblems), formatProblems(s.SkippedProblems))
+}
+
+func formatProblems(problems []string) string {
```

Comment:
We could make an alias type `type Slugs []string` and implement a `String()` on that which has this behavior. Then both `FetchedProblems` and `SkippedProblems` could be `Slugs`.